timelock-server/src/suiteTest/java/com/palantir/atlasdb/timelock/MultiNodePaxosTimeLockServerIntegrationTest.java

```diff
@@ -362,6 +365,41 @@ public void directLegacyAndConjureLockServicesInteractCorrectly() throws Interru
                 .isFalse();
     }
 
+    @Test
+    public void lockAcquiredByConjureLockServiceIsAlsoAcquiredInLegacy() throws InterruptedException {
+        com.palantir.lock.LockRequest lockRequest = com.palantir.lock.LockRequest.builder(
+                ImmutableSortedMap.<LockDescriptor, LockMode>naturalOrder()
+                        .put(StringLockDescriptor.of("lock"), LockMode.WRITE)
+                        .build())
+                .doNotBlock()
+                .build();
+        String anonymousId = LockClient.ANONYMOUS.getClientId();
+        ConjureLockV1Request conjureLockRequest = ConjureLockV1Request.builder()
+                .lockClient(anonymousId)
+                .lockRequest(lockRequest)
+                .build();
+
+        HeldLocksToken token = client.conjureLegacyLockService().lockAndGetHeldLocks(
+                AuthHeader.valueOf("Bearer unused"),
+                client.namespace(),
+                conjureLockRequest)
+                .orElseThrow(() -> new RuntimeException("We should have been able to get the lock"));
+
+        assertThat(client.legacyLockService().lockAndGetHeldLocks(anonymousId, lockRequest))
+                .as("if the conjure impl has taken a lock, the legacy impl mustn't be able to take it")
+                .isNull();
+        assertThat(client.legacyLockService().unlock(token.getLockRefreshToken())).isTrue();
+        assertThat(client.legacyLockService().lockAndGetHeldLocks(anonymousId, lockRequest))
+                .as("a lock once unlocked by conjure impl can be acquired by legacy impl")
```

Comment:
not unlocked by conjure impl

Reply:
oops 🔪 yeah, `lock taken by conjure impl that was unlocked, can now be acquired by legacy impl`